### PR TITLE
Lab2: Bubble Sort

### PR DESCRIPTION
**Pull request by ShahNikita19**
Fixes #42 

#### Changes proposed in this PR:

- Implementation for Bubble Sort using go lang
- Supported Test case implementation

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -2,6 +2,7 @@ github.com/davecgh/go-spew v1.1.0 h1:ZDRjVQ15GmhC3fiQ8ni8+OwkZQO4DARzQgrnXU1Liz8
 github.com/davecgh/go-spew v1.1.0/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSsI+c5H38=
```
Comment (mohankrishna) [inline go.sum]: Please revert this file

**Comment by mohankrishna** [comment]
Please check your folder structure. The folder structure should be `02_bubble/ShahNikita19` please create a new folder inside 02_bubble with name `ShahNikita19`and move the code files inside that. The main idea is to avoid any conflicts when everyone merge their PRs.

**Comment by anzdaddy** [comment]
Needs a more descriptive title.

File: 02_bubble/bubble.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+func bubbleSort(array []int) []int {
+	arrayLength := len(array)
+	for i := 0; i < arrayLength; i++ {
+		for j := 0; j < arrayLength-i-1; j++ {
+			if array[j] > array[j+1] {
+				array[j], array[j+1] = array[j+1], array[j]
+			}
+		}
+	}
+	return array
```
Comment (anzdaddy) [inline 02_bubble/bubble.go]: Can't mutate the original. Make a copy. But first write a test that detects the lack of copying.

File: 02_bubble/bubble_test.go [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+import (
+	"bytes"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestMainOutput(t *testing.T) {
```
Comment (juliaogris) [inline 02_bubble/bubble_test.go]: Please add `TestBubbleSort` comparing returned []int slice to expected int slice.   
Include corner cases such as empty slice, one element slice, and duplicated elements.

**Review by juliaogris — COMMENTED** [review]
I've updated your PR description to render as nice Markdown needs to be `- ` (space after dash) for bulletins

-   Please edit your commits to follow [recommendations](https://chris.beams.io/posts/git-commit/)
    -   Capitalize the subject line
    -   Use the imperative mood in the subject line
    -   Fold fixup commits into previous ones with `git rebase -i HASH`

File: go.sum (hunk 1 above)
Comment (ShahNikita19) [reply go.sum]: Done

File: 02_bubble/bubble.go (hunk 2 above)
Comment (ShahNikita19) [reply 02_bubble/bubble.go]: Done

File: 02_bubble/bubble_test.go (hunk 3 above)
Comment (ShahNikita19) [reply 02_bubble/bubble_test.go]: Done

File: 02_bubble/ShahNikita19/bubble.go [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+	fmt.Fprintln(out, bubbleSort([]int{3, 2, 1, 5}))
+}
+
+func bubbleSort(array []int) []int {
+	arrayLength := len(array)
+	copyArray := make([]int, len(array))
+	copy(copyArray, array)
+	for i := 0; i < arrayLength; i++ {
+		for j := 0; j < arrayLength-i-1; j++ {
+			if array[j] > array[j+1] {
```
Comment (anzdaddy) [inline 02_bubble/ShahNikita19/bubble.go]: Still mutating the original.

**Review by anzdaddy — CHANGES_REQUESTED** [review]

File: 02_bubble/ShahNikita19/bubble.go (hunk 4 above)
Comment (anzdaddy) [inline 02_bubble/ShahNikita19/bubble.go]: Before fixing this, write a test to detect the problem.

**Review by anzdaddy — CHANGES_REQUESTED** [review]

**Comment by juliaogris** [comment]
Click `Update branch` and locally `git pull origin lab2` before adding new commits.

**Comment by juliaogris** [comment]
Hi,
There has been  no activity in the last 20 days so I will close this PR.
Feel free to open it again if you have something to add.
Thanks!
Julia

**Pull request closed without merging**